detectron2/structures/image_list.py

```diff
@@ -70,16 +70,21 @@ def from_tensors(
             assert isinstance(t, torch.Tensor), type(t)
             assert t.shape[1:-2] == tensors[0].shape[1:-2], t.shape
         # per dimension maximum (H, W) or (C_1, ..., C_K, H, W) where K >= 1 among all tensors
-        max_size = tuple(max(s) for s in zip(*[img.shape for img in tensors]))
+        max_size = (
+            torch.stack(
+                [
+                    # Keep original tensor for tracing.
+                    torch.stack(size) if torch.is_tensor(size[0]) else torch.tensor(size)
+                    for size in [tuple(img.shape) for img in tensors]
+                ]
+            )
+            .max(0)
+            .values
+        )
```

Comment:
how about 
```
max_size = torch.as_tensor([img.shape for img in tensors]).max(0).values
```
?

Reply:
This will failed my unittest in 2 ways:
1. PyTorch:
- Normally:
  - `.shape` => `torch.Size`
  - `.shape[I]` => `int`
- During tracing:
  - `.shape` => `torch.Size`
  - `.shape[i]` => `torch.Tensor`
2. `as_tensor(List)` recreates a tensor with constants while `stack(List)` add a node in computation graph.

But your suggestion is very helpful.
I can replace the if-else with `torch.stack([torch.as_tensor(dim) for dim in size])`.